test/extensions/transport_sockets/tls/ecdsa_private_key_method_provider.cc

```diff
@@ -0,0 +1,180 @@
+#include "test/extensions/transport_sockets/tls/ecdsa_private_key_method_provider.h"
+
+#include <memory>
+
+#include "envoy/api/api.h"
+
+#include "openssl/ssl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace PrivateKeyMethodProvider {
+
+int EcdsaPrivateKeyMethodProvider::ssl_ecdsa_connection_index = -1;
+
+void EcdsaPrivateKeyConnection::delayed_op() {
+  const std::chrono::milliseconds timeout_0ms{0};
+
+  timer_ = dispatcher_.createTimer([this]() -> void {
+    finished_ = true;
+    this->cb_.complete();
+    return;
+  });
+  timer_->enableTimer(timeout_0ms);
+}
+
+static ssl_private_key_result_t privateKeySign(SSL* ssl, uint8_t* out, size_t* out_len,
+                                               size_t max_out, uint16_t signature_algorithm,
+                                               const uint8_t* in, size_t in_len) {
+  (void)out;
+  (void)out_len;
+  (void)signature_algorithm;
+  unsigned char hash[EVP_MAX_MD_SIZE];
```

Comment:
(Seems really odd to declare all variables at the top, C89-style, but maybe Envoy does this?)